Console/Command/Indexer/AbstractIndexerCommand.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Algolia\AlgoliaSearch\Console\Command\Indexer;
+
+use Algolia\AlgoliaSearch\Console\Command\AbstractStoreCommand;
+use Algolia\AlgoliaSearch\Service\StoreNameFetcher;
+use Magento\Framework\App\State;
+use Magento\Store\Model\StoreManagerInterface;
+
+abstract class AbstractIndexerCommand extends AbstractStoreCommand
+{
+    public function __construct(
+        protected StoreManagerInterface $storeManager,
+        State $state,
+        StoreNameFetcher $storeNameFetcher,
+        ?string $name = null
+    ) {
+        parent::__construct($state, $storeNameFetcher, $name);
+    }
+
+    protected function getCommandPrefix(): string
+    {
+        return parent::getCommandPrefix() . 'reindex:';
```

Comment:
We're going to need a task to update our [documentation](https://www.algolia.com/doc/integration/magento-2/how-it-works/indexing/?client=php#manual-indexing) for these new CLIs.

Reply:
task created (MAGE-1225)